helm-chart/templates/cluster.yaml

```diff
@@ -5,6 +5,12 @@ metadata:
   name: {{ .Release.Name }}-cluster
   labels:
     app: {{ .Release.Name }}
+    annotations:
+      # We use the name of the CR as external name to cover the cases where
+      # the default naming scheme (<namespace>-<name>-<5 rand char>) turns out
+      # to be too long for some Kubernetes providers like GKE who has max 40
+      # chars limit.
+      crossplane.io/external-name: {{ .Release.Name }}
```

Comment:
do you think it's worth also adding a note in the comment here for why we can't use the helm functions like rand and trunc? i didn't realize that until I went back to the issue and saw your comment: https://github.com/crossplane/provider-gcp/issues/222#issuecomment-620262248